src/mesh/mesh.cc

```diff
@@ -777,6 +777,45 @@ void Mesh::make_edge_permutations()
 			break;
 		}
 	}
+
+	for (vector<BoundaryData>::iterator bdr=boundary_.begin(); bdr!=boundary_.end(); bdr++)
+	{
+        if (bdr->bc_ele_idx_ >= element_vec_.size()) continue; // skip invalid boundary item
```

Comment:
Shouldn't here be '<' in place of '>=' ?

Reply:
See method Mesh::make_neighbours_and_edges. There are two main for cycles. We add boundary elements to mesh (not loaded from mesh file) in second of them. There is some 'hack' that adds two items to boundary_ vector for every new boundary element (but use only one item). We must skip unused (invalid) items in marked line.